packages/framework/src/Component/Elasticsearch/IndexRepository.php

```diff
@@ -0,0 +1,262 @@
+     */
+    public function deleteIndex(string $indexName): void
+    {
+        $indexes = $this->elasticsearchClient->indices();
+
+        $result = $indexes->delete([
+            'index' => $indexName,
+        ]);
+
+        if (isset($result['error'])) {
+            throw ElasticsearchIndexException::deleteIndexError($indexName, $result['error']);
+        }
+    }
+
+    /**
+     * @param string $indexAlias
+     * @param int $domainId
+     * @param array $ids
+     */
+    public function deleteIds(string $indexAlias, int $domainId, array $ids): void
+    {
+        $this->elasticsearchClient->deleteByQuery([
+            'index' => $indexAlias,
+            'type' => '_doc',
+            'body' => [
+                'query' => [
+                    'bool' => [
+                        'must' => [
+                            'ids' => [
+                                'values' => array_values($ids),
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+        ]);
+    }
+
+    /**
+     * @param string $indexAlias
+     * @param array $data
+     * @param bool $createIfNotExists
+     */
+    public function bulkUpdate(string $indexAlias, array $data, bool $createIfNotExists = true): void
+    {
+        $params = ['body' => []];
+
+        foreach ($data as $id => $row) {
+            $params['body'][] = [
+                'update' => [
+                    '_index' => $indexAlias,
+                    '_type' => '_doc',
+                    '_id' => (string)$id,
+                ],
+            ];
+
+            $params['body'][] = [
+                'doc' => $row,
+                'doc_as_upsert' => $createIfNotExists,
+            ];
+        }
+
+        $result = $this->elasticsearchClient->bulk($params);
+
+        if (isset($result['errors']) && $result['errors'] === true) {
+            throw ElasticsearchIndexException::bulkUpdateError($indexAlias, $result['items']);
+        }
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Elasticsearch\IndexDefinition $indexDefinition
+     * @param array $restrictToIds
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     */
+    public function export(IndexDefinition $indexDefinition, array $restrictToIds, OutputInterface $output): void
+    {
+        $indexAlias = $indexDefinition->getIndexAlias();
+        $index = $indexDefinition->getIndex();
+        $domainId = $indexDefinition->getDomainId();
+
+        if ($restrictToIds !== []) {
+            $totalCount = count($restrictToIds);
+        } else {
+            $totalCount = $index->getTotalCount($indexDefinition->getDomainId());
+        }
+
+        $progressBar = $this->progressBarFactory->create($output, $totalCount);
+
+        $lastProcessedId = 0;
+        $updatedIds = [];
+        do {
+            $currentBatchData = $index->getExportData($domainId, $lastProcessedId, $restrictToIds);
```

Comment:
I was a bit lost, on this method. it exports data, but the last two arguments are mutually exclusive (= it doesn't make sense to set them both).
I'm thinking about split the whole export method into two separate (batch and ids). What do you think? 
I believe it could help with readability.